rust_src/src/keyboard.rs

```diff
@@ -32,6 +34,64 @@ use crate::{
 #[cfg(feature = "window-system")]
 use crate::remacs_sys::cancel_hourglass;
 
+#[derive(Clone, Copy, Debug)]
+pub struct Event(LispObject);
+
```

Comment:
Several of these functions would be more efficient as `match` against as_cons instead of using if and then force. Also, this style would make for easier debugging.

Reply:
Fixed.